coral-incremental/src/main/java/com/linkedin/coral/incremental/RelNodeIncrementalTransformer.java

```diff
@@ -49,61 +66,130 @@ public RelNode visit(TableScan scan) {
       public RelNode visit(LogicalJoin join) {
         RelNode left = join.getLeft();
         RelNode right = join.getRight();
-        RelNode incrementalLeft = convertRelIncremental(left);
-        RelNode incrementalRight = convertRelIncremental(right);
+        IncrementalTransformerResults incrementalTransformerResultsLeft = convertRelIncremental(left);
+        IncrementalTransformerResults incrementalTransformerResultsRight = convertRelIncremental(right);
+        RelNode incrementalLeft = incrementalTransformerResultsLeft.getIncrementalRelNode();
+        RelNode incrementalRight = incrementalTransformerResultsRight.getIncrementalRelNode();
+        incrementalTransformerResults
```

Comment:
will it simplify the logic if we make `intermediateQueryRelNodes` global?

Reply:
Yes, we are actually getting rid of IncrementalTransformerResults and keeping all logic inside of the transformer itself (larger discussion in the Slack channel).